src/python/gudhi/sklearn/rips_persistence.py

```diff
@@ -0,0 +1,148 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Vincent Rouvreau
+#
+# Copyright (C) 2022 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+from .. import RipsComplex
+from sklearn.base import BaseEstimator, TransformerMixin
+
+import numpy as np
+# joblib is required by scikit-learn
+from joblib import Parallel, delayed
+
+# Mermaid sequence diagram - https://mermaid-js.github.io/mermaid-live-editor/
+# sequenceDiagram
+#     USER->>RipsPersistence: fit_transform(X)
+#     Note right of RipsPersistence: homology_dimensions=[i,j]
+#     RipsPersistence->>thread1: _tranform(X[0])
+#     RipsPersistence->>thread2: _tranform(X[1])
+#     Note right of RipsPersistence: ...
+#     thread1->>RipsPersistence: [array( Hi(X[0]) ), array( Hj(X[0]) )]
+#     thread2->>RipsPersistence: [array( Hi(X[1]) ), array( Hj(X[1]) )]
+#     Note right of RipsPersistence: ...
+#     RipsPersistence->>USER: [[array( Hi(X[0]) ), array( Hj(X[0]) )],<br/> [array( Hi(X[1]) ), array( Hj(X[1]) )],<br/> ...]
+
+
+class RipsPersistence(BaseEstimator, TransformerMixin):
+    """
+    This is a class for computing the persistence diagrams from a Vietoris-Rips complex.
+    """
+
+    def __init__(
+        self,
+        homology_dimensions,
+        max_edge_length=float('inf'),
+        input_type='point cloud',
+        nb_collapse=-1,
+        homology_coeff_field=11,
+        min_persistence=0.0,
+        expand_extra_dimension = True,
+        n_jobs=None,
+    ):
+        """
+        Constructor for the RipsPersistence class.
+
+        Parameters:
+            homology_dimensions (int or list of int): The returned persistence diagrams dimension(s).
+                Short circuit the use of :class:`~gudhi.representations.preprocessing.DimensionSelector` when only one
+                dimension matters (in other words, when `homology_dimensions` is an int).
+            max_edge_length (float): Rips value. Default is +Inf.
+            input_type (str): Can be 'point cloud' when inputs are point clouds, or 'lower distance matrix', when
+                inputs are lower triangular distance matrix (can be full square, but the upper part of the distance
+                matrix will not be considered). Default is 'point cloud'.
+            nb_collapse (int): The number of :func:`~gudhi.SimplexTree.collapse_edges` iterations to perform on the
```

Comment:
`nb` for number is not consistent with gudhi or e.g. numpy (`st.num_simplices`, ...). Can we change `nb_collapse` to `num_collapses` ?

Reply:
Currently we have [nb_iterations](https://gudhi.inria.fr/python/latest/simplex_tree_ref.html#gudhi.SimplexTree.collapse_edges.params.nb_iterations) in the SimplexTree. And `n_jobs` which uses neither `num_` nor `nb_`...
I don't have a strong opinion on this name. Both can be a bit confusing in that they count the number of passes of the collapse algorithm, not the number of edges that are collapsed.

Reply:
I renamed `nb_collapse` as `num_collapses` that can be an integer or a bool. Default value is True aka. "automatic mode"
cf. f96eccc58